stdlib/logging/__init__.pyi

```diff
@@ -63,125 +63,160 @@ class Logger(Filterer):
     if sys.version_info >= (3, 8):
         def debug(
             self,
-            msg: Any,
-            *args: Any,
+            msg: object,
+            *args: object,
             exc_info: _ExcInfoType = ...,
             stack_info: bool = ...,
             stacklevel: int = ...,
-            extra: dict[str, Any] | None = ...,
+            extra: dict[str, object] | None = ...,
         ) -> None: ...
         def info(
             self,
-            msg: Any,
-            *args: Any,
+            msg: object,
+            *args: object,
             exc_info: _ExcInfoType = ...,
             stack_info: bool = ...,
             stacklevel: int = ...,
-            extra: dict[str, Any] | None = ...,
+            extra: dict[str, object] | None = ...,
         ) -> None: ...
         def warning(
             self,
-            msg: Any,
-            *args: Any,
+            msg: object,
+            *args: object,
             exc_info: _ExcInfoType = ...,
             stack_info: bool = ...,
             stacklevel: int = ...,
-            extra: dict[str, Any] | None = ...,
+            extra: dict[str, object] | None = ...,
         ) -> None: ...
         def warn(
             self,
-            msg: Any,
-            *args: Any,
+            msg: object,
+            *args: object,
             exc_info: _ExcInfoType = ...,
             stack_info: bool = ...,
             stacklevel: int = ...,
-            extra: dict[str, Any] | None = ...,
+            extra: dict[str, object] | None = ...,
```

Comment:
This doesn't work for dicts, because dicts are invariant: a `dict[str, int]` is not a valid `dict[str, object]`. The reason is that if `foo` is a `dict[str, object]`, then `foo["bar"] = "lol"` is valid, but that really shouldn't be valid if  `foo` is a `dict[str, int]`.

`Any` is really needed for this: we want to say "do not complain about the value type, whatever type it is" (`Any`), not "the values can be any objects" (`object`).

Reply:
Right, I missed that, thanks. However, I don't think there's a need for the invariant `dict` here, is there? Replacing this with the covariant `Mapping` should work, shouldn't it?

Reply:
Yes, it can be `Mapping`, or even one of the mapping-like `SupportsFooAndBar` protocols in `_typeshed`. The `extra` dict gets passed to `makeRecord`, so it can be whatever `makeRecord` supports.